{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6971bbd5-1bd6-4572-a03d-4c360100b51c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import gzip\n",
    "\n",
    "def clinvar_vcf_to_pd(vcf_path):\n",
    "    # list where we will store dictionary params of each variant\n",
    "    variants_params = list()\n",
    "    with gzip.open(vcf_path, \"rt\") as file:\n",
    "        for line in file:\n",
    "            # dictinary where we will store each parameter of the variant\n",
    "            if line.startswith(\"#\"):\n",
    "                # descriptor lines not interested in\n",
    "                continue\n",
    "            fields = line.split(\"\\t\")\n",
    "            # dictinary where we will store each parameter of the variant\n",
    "            # obtaining parameters from each variant \n",
    "            chrom = fields[0]\n",
    "            pos = fields[1]\n",
    "            id = fields[2]\n",
    "            ref = fields[3]\n",
    "            alt = fields[4]\n",
    "            qual = fields[5]\n",
    "            filter = fields[6]\n",
    "            info = fields[7]\n",
    "            dict_params = {\n",
    "                \"Chrom\" : chrom,\n",
    "                \"Pos\" : pos,\n",
    "                \"Id\" : id,\n",
    "                \"Ref\" : ref,\n",
    "                \"Alt\" : alt,\n",
    "                \"Qual\" : qual,\n",
    "                \"Filter\" : filter,\n",
    "\n",
    "            }\n",
    "\n",
    "            # in info we have different parameters \n",
    "            clnv_params = info.split(\";\")\n",
    "\n",
    "            for clnv_param in clnv_params:\n",
    "                key_value = clnv_param.split(\"=\")\n",
    "                key = key_value[0]\n",
    "                value = key_value[1]\n",
    "                # it's a comma seperated list of molecular consequences\n",
    "                if key == \"MC\":\n",
    "                    if \",\" in value:\n",
    "                        mol_conseqs_ids = list()\n",
    "                        mol_conseqs = value.split(\",\")\n",
    "                        # print(mol_conseqs)\n",
    "                        for mol_conseq in mol_conseqs:\n",
    "                            # taking the string id to be converted into factors \n",
    "                            # print(mol_conseq)\n",
    "                            mol_conseq_id = mol_conseq.split(\"|\")[1]\n",
    "                            mol_conseqs_ids.append(mol_conseq_id)\n",
    "                        value = \",\".join(mol_conseqs_ids)\n",
    "                    else:\n",
    "                        value = value.split(\"|\")[1]\n",
    "\n",
    "\n",
    "                dict_params[key] = value\n",
    "            \n",
    "\n",
    "            variants_params.append(dict_params)\n",
    "    return(variants_params)\n",
    "\n",
    "\n",
    "variants_params = clinvar_vcf_to_pd(\"/home/ocanal/ANN_DIR/clinvar/hg38/clinvar_20231104.vcf.gz\")\n",
    "df = pd.DataFrame(variants_params)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1ea2993d-6af3-4ec5-8518-e7c577e193b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Id</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Qual</th>\n",
       "      <th>Filter</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>...</th>\n",
       "      <th>RS</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>CLNSIGCONF</th>\n",
       "      <th>AF_TGP</th>\n",
       "      <th>CLNVI</th>\n",
       "      <th>CLNDISDBINCL</th>\n",
       "      <th>CLNDNINCL</th>\n",
       "      <th>CLNSIGINCL</th>\n",
       "      <th>DBVARID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2193183</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2238986</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2386655</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2278803</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2333177</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300616</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273806</td>\n",
       "      <td>2219599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2206917</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300617</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>273866</td>\n",
       "      <td>2237818</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2232003</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300618</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>274366</td>\n",
       "      <td>2206666</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2200058</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300619</th>\n",
       "      <td>NT_187693.1</td>\n",
       "      <td>275068</td>\n",
       "      <td>2241971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2226217</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300620</th>\n",
       "      <td>NW_009646201.1</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>32774</td>\n",
       "      <td>Human_Phenotype_Ontology:HP:0032224,MedGen:C00...</td>\n",
       "      <td>ABO_blood_group_system|not_provided</td>\n",
       "      <td>...</td>\n",
       "      <td>1556058284\\n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OMIM:110300.0001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2300621 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Chrom     Pos       Id Ref Alt Qual Filter ALLELEID  \\\n",
       "0                     1   69134  2205837   A   G    .      .  2193183   \n",
       "1                     1   69581  2252161   C   G    .      .  2238986   \n",
       "2                     1   69682  2396347   G   A    .      .  2386655   \n",
       "3                     1   69769  2288999   T   C    .      .  2278803   \n",
       "4                     1   69995  2351346   G   C    .      .  2333177   \n",
       "...                 ...     ...      ...  ..  ..  ...    ...      ...   \n",
       "2300616     NT_187693.1  273806  2219599   G   A    .      .  2206917   \n",
       "2300617     NT_187693.1  273866  2237818   A   C    .      .  2232003   \n",
       "2300618     NT_187693.1  274366  2206666   G   C    .      .  2200058   \n",
       "2300619     NT_187693.1  275068  2241971   T   C    .      .  2226217   \n",
       "2300620  NW_009646201.1   83614    17735  TC   T    .      .    32774   \n",
       "\n",
       "                                                  CLNDISDB  \\\n",
       "0                             MeSH:D030342,MedGen:C0950123   \n",
       "1                             MeSH:D030342,MedGen:C0950123   \n",
       "2                             MeSH:D030342,MedGen:C0950123   \n",
       "3                             MeSH:D030342,MedGen:C0950123   \n",
       "4                             MeSH:D030342,MedGen:C0950123   \n",
       "...                                                    ...   \n",
       "2300616                       MeSH:D030342,MedGen:C0950123   \n",
       "2300617                       MeSH:D030342,MedGen:C0950123   \n",
       "2300618                       MeSH:D030342,MedGen:C0950123   \n",
       "2300619                       MeSH:D030342,MedGen:C0950123   \n",
       "2300620  Human_Phenotype_Ontology:HP:0032224,MedGen:C00...   \n",
       "\n",
       "                                       CLNDN  ...            RS AF_EXAC  \\\n",
       "0                    Inborn_genetic_diseases  ...           NaN     NaN   \n",
       "1                    Inborn_genetic_diseases  ...           NaN     NaN   \n",
       "2                    Inborn_genetic_diseases  ...           NaN     NaN   \n",
       "3                    Inborn_genetic_diseases  ...           NaN     NaN   \n",
       "4                    Inborn_genetic_diseases  ...           NaN     NaN   \n",
       "...                                      ...  ...           ...     ...   \n",
       "2300616              Inborn_genetic_diseases  ...           NaN     NaN   \n",
       "2300617              Inborn_genetic_diseases  ...           NaN     NaN   \n",
       "2300618              Inborn_genetic_diseases  ...           NaN     NaN   \n",
       "2300619              Inborn_genetic_diseases  ...           NaN     NaN   \n",
       "2300620  ABO_blood_group_system|not_provided  ...  1556058284\\n     NaN   \n",
       "\n",
       "        AF_ESP CLNSIGCONF AF_TGP             CLNVI CLNDISDBINCL CLNDNINCL  \\\n",
       "0          NaN        NaN    NaN               NaN          NaN       NaN   \n",
       "1          NaN        NaN    NaN               NaN          NaN       NaN   \n",
       "2          NaN        NaN    NaN               NaN          NaN       NaN   \n",
       "3          NaN        NaN    NaN               NaN          NaN       NaN   \n",
       "4          NaN        NaN    NaN               NaN          NaN       NaN   \n",
       "...        ...        ...    ...               ...          ...       ...   \n",
       "2300616    NaN        NaN    NaN               NaN          NaN       NaN   \n",
       "2300617    NaN        NaN    NaN               NaN          NaN       NaN   \n",
       "2300618    NaN        NaN    NaN               NaN          NaN       NaN   \n",
       "2300619    NaN        NaN    NaN               NaN          NaN       NaN   \n",
       "2300620    NaN        NaN    NaN  OMIM:110300.0001          NaN       NaN   \n",
       "\n",
       "        CLNSIGINCL DBVARID  \n",
       "0              NaN     NaN  \n",
       "1              NaN     NaN  \n",
       "2              NaN     NaN  \n",
       "3              NaN     NaN  \n",
       "4              NaN     NaN  \n",
       "...            ...     ...  \n",
       "2300616        NaN     NaN  \n",
       "2300617        NaN     NaN  \n",
       "2300618        NaN     NaN  \n",
       "2300619        NaN     NaN  \n",
       "2300620        NaN     NaN  \n",
       "\n",
       "[2300621 rows x 28 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4e149792-0346-4459-babe-19124eda5f90",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CLNSIG\n",
       "Uncertain_significance                                            1119929\n",
       "Likely_benign                                                      606654\n",
       "Benign                                                             193415\n",
       "Pathogenic                                                         135246\n",
       "Conflicting_interpretations_of_pathogenicity                       103898\n",
       "                                                                   ...   \n",
       "Conflicting_interpretations_of_pathogenicity|association|other          1\n",
       "Uncertain_risk_allele|protective                                        1\n",
       "Affects|association                                                     1\n",
       "other|risk_factor                                                       1\n",
       "Likely_benign|risk_factor                                               1\n",
       "Name: count, Length: 88, dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"CLNSIG\"].value_counts()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d9cb3b73-2743-435a-b12a-d6207837b9e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Likely_benign', 'Uncertain_significance', 'Benign',\n",
       "       'Conflicting_interpretations_of_pathogenicity', 'Pathogenic',\n",
       "       'Likely_pathogenic', 'Benign/Likely_benign', 'not_provided',\n",
       "       'Pathogenic/Likely_pathogenic', nan, 'risk_factor', 'Affects',\n",
       "       'association', 'Benign|other',\n",
       "       'Conflicting_interpretations_of_pathogenicity|other',\n",
       "       'drug_response',\n",
       "       'Conflicting_interpretations_of_pathogenicity|association',\n",
       "       'Uncertain_risk_allele', 'other',\n",
       "       'Uncertain_significance|risk_factor',\n",
       "       'Likely_pathogenic|risk_factor', 'Likely_benign|association',\n",
       "       'Likely_risk_allele', 'Pathogenic/Likely_pathogenic|other',\n",
       "       'Pathogenic|other',\n",
       "       'Pathogenic/Likely_pathogenic/Pathogenic,_low_penetrance',\n",
       "       'Pathogenic/Likely_pathogenic/Pathogenic,_low_penetrance|other',\n",
       "       'protective', 'Pathogenic|risk_factor',\n",
       "       'Pathogenic/Likely_pathogenic|risk_factor',\n",
       "       'Benign/Likely_benign|risk_factor',\n",
       "       'Uncertain_significance/Uncertain_risk_allele',\n",
       "       'Pathogenic|association|protective', 'protective|risk_factor',\n",
       "       'Benign/Likely_benign|other|risk_factor',\n",
       "       'Benign/Likely_benign|association', 'Benign|association',\n",
       "       'Affects|association|other', 'Pathogenic|protective',\n",
       "       'Likely_pathogenic,_low_penetrance',\n",
       "       'Conflicting_interpretations_of_pathogenicity|drug_response|other',\n",
       "       'Conflicting_interpretations_of_pathogenicity|drug_response',\n",
       "       'Benign|drug_response', 'Likely_pathogenic|other',\n",
       "       'Conflicting_interpretations_of_pathogenicity|association|other',\n",
       "       'Conflicting_interpretations_of_pathogenicity|protective',\n",
       "       'Pathogenic/Likely_pathogenic|drug_response',\n",
       "       'Benign/Likely_benign|other',\n",
       "       'Likely_pathogenic/Likely_risk_allele',\n",
       "       'Uncertain_risk_allele|protective', 'association_not_found',\n",
       "       'Affects|association', 'Uncertain_significance|association',\n",
       "       'Likely_benign|other', 'Uncertain_significance|other',\n",
       "       'Conflicting_interpretations_of_pathogenicity|association|risk_factor',\n",
       "       'Pathogenic|association', 'Benign|risk_factor',\n",
       "       'Conflicting_interpretations_of_pathogenicity|other|risk_factor',\n",
       "       'Pathogenic/Likely_risk_allele|risk_factor',\n",
       "       'drug_response|risk_factor',\n",
       "       'Uncertain_significance|drug_response',\n",
       "       'Conflicting_interpretations_of_pathogenicity|risk_factor',\n",
       "       'other|risk_factor',\n",
       "       'Pathogenic/Likely_pathogenic/Likely_risk_allele',\n",
       "       'Likely_pathogenic|drug_response',\n",
       "       'Conflicting_interpretations_of_pathogenicity|Affects',\n",
       "       'association|drug_response|risk_factor', 'Pathogenic|Affects',\n",
       "       'Pathogenic|drug_response', 'Affects|risk_factor',\n",
       "       'Pathogenic|drug_response|other', 'confers_sensitivity',\n",
       "       'Likely_pathogenic|association', 'Benign|Affects',\n",
       "       'Likely_pathogenic|Affects', 'Uncertain_risk_allele|risk_factor',\n",
       "       'Pathogenic/Likely_risk_allele',\n",
       "       'Likely_benign|drug_response|other',\n",
       "       'Benign/Likely_benign|drug_response',\n",
       "       'Benign/Likely_benign|drug_response|other', 'drug_response|other',\n",
       "       'association|drug_response', 'Pathogenic|confers_sensitivity',\n",
       "       'association|risk_factor', 'Benign|confers_sensitivity',\n",
       "       'confers_sensitivity|other',\n",
       "       'Likely_pathogenic/Pathogenic,_low_penetrance',\n",
       "       'Likely_benign|risk_factor'], dtype=object)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"CLNSIG\"].unique()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
