{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6971bbd5-1bd6-4572-a03d-4c360100b51c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import gzip\n",
    "\n",
    "def clinvar_vcf_to_pd(vcf_path: str) -> list:\n",
    "    \"\"\"\n",
    "    From a vcf file it creates a list containing information of each mutation\n",
    "    described in clinvar as a dictionary.\n",
    "    Args:\n",
    "        vcf_path (str): path to the vcf file where we will extract the information\n",
    "    Returns:\n",
    "        variants_params (list): list of dictionaries where each item of a list is a dictionary\n",
    "            that describes the variant\n",
    "    \"\"\"\n",
    "    \n",
    "    # list where we will store dictionary params of each variant\n",
    "    variants_params = list()\n",
    "    with gzip.open(vcf_path, \"rt\") as file:\n",
    "        for line in file:\n",
    "            # remove \\n\n",
    "            line = line.strip()\n",
    "            # dictinary where we will store each parameter of the variant\n",
    "            if line.startswith(\"#\"):\n",
    "                # descriptor lines not interested in\n",
    "                continue\n",
    "            fields = line.split(\"\\t\")\n",
    "            # dictinary where we will store each parameter of the variant\n",
    "            # obtaining parameters from each variant \n",
    "            chrom = fields[0]\n",
    "            pos = fields[1]\n",
    "            id = fields[2]\n",
    "            ref = fields[3]\n",
    "            alt = fields[4]\n",
    "            qual = fields[5]\n",
    "            filter = fields[6]\n",
    "            info = fields[7]\n",
    "            dict_params = {\n",
    "                \"Chrom\" : chrom,\n",
    "                \"Pos\" : pos,\n",
    "                \"Id\" : id,\n",
    "                \"Ref\" : ref,\n",
    "                \"Alt\" : alt,\n",
    "                \"Qual\" : qual,\n",
    "                \"Filter\" : filter,\n",
    "\n",
    "            }\n",
    "            # print(fields)\n",
    "\n",
    "            # in info we have different parameters \n",
    "            clnv_params = info.split(\";\")\n",
    "\n",
    "            for clnv_param in clnv_params:\n",
    "                key_value = clnv_param.split(\"=\")\n",
    "                key = key_value[0]\n",
    "                value = key_value[1]\n",
    "\n",
    "                # add each annotation as a key_dict value in dict_params    \n",
    "                dict_params[key] = value\n",
    "            \n",
    "            # create a list of  mutations with dicts that store its information\n",
    "            variants_params.append(dict_params)\n",
    "                \n",
    "    return(variants_params)\n",
    "\n",
    "\n",
    "def clean_clinvar_file(clinvar_path, cleaned_file):\n",
    "    \n",
    "    # we just want mutations that are classified as the following items\n",
    "    sig = [\"Pathogenic\", \"Benign\", \"Likely_pathogenic\", \"Likely_benign\", \"Uncertain_significance\"]\n",
    "    \n",
    "    with gzip.open(clinvar_path, \"rt\", encoding='utf-8') as file:\n",
    "        with open(cleaned_file, \"w\") as write_file:\n",
    "            for line in file:\n",
    "                # dictinary where we will store each parameter of the variant\n",
    "                if line.startswith(\"#\"):\n",
    "                    write_file.write(line)\n",
    "\n",
    "                    # descriptor lines not interested in\n",
    "                    continue\n",
    "                # It is a tab seperated file so we split it using tabs\n",
    "                fields = line.split(\"\\t\")\n",
    "                # all annotations are in info field which is positioned at 7\n",
    "                info = fields[7]\n",
    "\n",
    "                # clinvar annotations are seperated by ;\n",
    "                clnv_params = info.split(\";\")\n",
    "\n",
    "                for clnv_param in clnv_params:\n",
    "                    # each annotation is a key=value pair\n",
    "                    key_value = clnv_param.split(\"=\")\n",
    "                    key = key_value[0]\n",
    "                    value = key_value[1]\n",
    "                    # in CLNSIG is the pathogenicity of the mutation\n",
    "                    if key == \"CLNSIG\":\n",
    "                        # we only take main \n",
    "                        if any(sign in value for sign in sig):\n",
    "                            clin_sig = True\n",
    "                if clin_sig:\n",
    "                    write_file.write(line)\n",
    "# clean_clinvar_file(\"/home/ocanal/Desktop/clinvar/hg19/clinvar_20231104.vcf.gz\", \"/home/ocanal/Desktop/clinvar/hg19/clinvar_parsed_hg19.vcf\")\n",
    "                    \n",
    "\n",
    "variants_params = clinvar_vcf_to_pd(\"/home/ocanal/Desktop/clinvar/hg19/clinvar_20231104.vcf.gz\")\n",
    "df = pd.DataFrame(variants_params)\n",
    "# print(variants_params)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2f01a0cb-b43e-47c0-8a16-9d1bf583d520",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Id</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Qual</th>\n",
       "      <th>Filter</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>...</th>\n",
       "      <th>RS</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>CLNSIGCONF</th>\n",
       "      <th>AF_TGP</th>\n",
       "      <th>CLNVI</th>\n",
       "      <th>CLNDISDBINCL</th>\n",
       "      <th>CLNDNINCL</th>\n",
       "      <th>CLNSIGINCL</th>\n",
       "      <th>DBVARID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2193183</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2238986</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2386655</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2278803</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2333177</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300570</th>\n",
       "      <td>NW_003315925.1</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>32774</td>\n",
       "      <td>Human_Phenotype_Ontology:HP:0032224,MedGen:C00...</td>\n",
       "      <td>ABO_blood_group_system|not_provided</td>\n",
       "      <td>...</td>\n",
       "      <td>1556058284</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OMIM:110300.0001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300571</th>\n",
       "      <td>NW_003315947.1</td>\n",
       "      <td>181683</td>\n",
       "      <td>156304</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>166084</td>\n",
       "      <td>MedGen:C3272265|MedGen:CN517202|.</td>\n",
       "      <td>Three_Vessel_Coronary_Disease|not_provided|PLA...</td>\n",
       "      <td>...</td>\n",
       "      <td>281865545</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ClinGen:CA123258|OMIM:173445.0001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300572</th>\n",
       "      <td>NW_003315950.2</td>\n",
       "      <td>355765</td>\n",
       "      <td>206</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>15245</td>\n",
       "      <td>MONDO:MONDO:0013193,MedGen:C2750473,OMIM:61323...</td>\n",
       "      <td>Thyrotoxic_periodic_paralysis,_susceptibility_...</td>\n",
       "      <td>...</td>\n",
       "      <td>672601244</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ClinGen:CA114051|OMIM:613236.0003|UniProtKB:B7...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300573</th>\n",
       "      <td>NW_003315950.2</td>\n",
       "      <td>356212</td>\n",
       "      <td>205</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>15244</td>\n",
       "      <td>MONDO:MONDO:0013193,MedGen:C2750473,OMIM:61323...</td>\n",
       "      <td>Thyrotoxic_periodic_paralysis,_susceptibility_...</td>\n",
       "      <td>...</td>\n",
       "      <td>527236158</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ClinGen:CA114049|OMIM:613236.0002|UniProtKB:B7...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300574</th>\n",
       "      <td>NW_003315950.2</td>\n",
       "      <td>356248</td>\n",
       "      <td>207</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>15246</td>\n",
       "      <td>MONDO:MONDO:0013193,MedGen:C2750473,OMIM:61323...</td>\n",
       "      <td>Thyrotoxic_periodic_paralysis,_susceptibility_...</td>\n",
       "      <td>...</td>\n",
       "      <td>527236159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ClinGen:CA114053|OMIM:613236.0004|UniProtKB:B7...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2300575 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Chrom     Pos       Id Ref Alt Qual Filter ALLELEID  \\\n",
       "0                     1   69134  2205837   A   G    .      .  2193183   \n",
       "1                     1   69581  2252161   C   G    .      .  2238986   \n",
       "2                     1   69682  2396347   G   A    .      .  2386655   \n",
       "3                     1   69769  2288999   T   C    .      .  2278803   \n",
       "4                     1   69995  2351346   G   C    .      .  2333177   \n",
       "...                 ...     ...      ...  ..  ..  ...    ...      ...   \n",
       "2300570  NW_003315925.1   83614    17735  TC   T    .      .    32774   \n",
       "2300571  NW_003315947.1  181683   156304   C   G    .      .   166084   \n",
       "2300572  NW_003315950.2  355765      206   G   A    .      .    15245   \n",
       "2300573  NW_003315950.2  356212      205   C   T    .      .    15244   \n",
       "2300574  NW_003315950.2  356248      207   A   G    .      .    15246   \n",
       "\n",
       "                                                  CLNDISDB  \\\n",
       "0                             MeSH:D030342,MedGen:C0950123   \n",
       "1                             MeSH:D030342,MedGen:C0950123   \n",
       "2                             MeSH:D030342,MedGen:C0950123   \n",
       "3                             MeSH:D030342,MedGen:C0950123   \n",
       "4                             MeSH:D030342,MedGen:C0950123   \n",
       "...                                                    ...   \n",
       "2300570  Human_Phenotype_Ontology:HP:0032224,MedGen:C00...   \n",
       "2300571                  MedGen:C3272265|MedGen:CN517202|.   \n",
       "2300572  MONDO:MONDO:0013193,MedGen:C2750473,OMIM:61323...   \n",
       "2300573  MONDO:MONDO:0013193,MedGen:C2750473,OMIM:61323...   \n",
       "2300574  MONDO:MONDO:0013193,MedGen:C2750473,OMIM:61323...   \n",
       "\n",
       "                                                     CLNDN  ...          RS  \\\n",
       "0                                  Inborn_genetic_diseases  ...         NaN   \n",
       "1                                  Inborn_genetic_diseases  ...         NaN   \n",
       "2                                  Inborn_genetic_diseases  ...         NaN   \n",
       "3                                  Inborn_genetic_diseases  ...         NaN   \n",
       "4                                  Inborn_genetic_diseases  ...         NaN   \n",
       "...                                                    ...  ...         ...   \n",
       "2300570                ABO_blood_group_system|not_provided  ...  1556058284   \n",
       "2300571  Three_Vessel_Coronary_Disease|not_provided|PLA...  ...   281865545   \n",
       "2300572  Thyrotoxic_periodic_paralysis,_susceptibility_...  ...   672601244   \n",
       "2300573  Thyrotoxic_periodic_paralysis,_susceptibility_...  ...   527236158   \n",
       "2300574  Thyrotoxic_periodic_paralysis,_susceptibility_...  ...   527236159   \n",
       "\n",
       "        AF_EXAC AF_ESP CLNSIGCONF AF_TGP  \\\n",
       "0           NaN    NaN        NaN    NaN   \n",
       "1           NaN    NaN        NaN    NaN   \n",
       "2           NaN    NaN        NaN    NaN   \n",
       "3           NaN    NaN        NaN    NaN   \n",
       "4           NaN    NaN        NaN    NaN   \n",
       "...         ...    ...        ...    ...   \n",
       "2300570     NaN    NaN        NaN    NaN   \n",
       "2300571     NaN    NaN        NaN    NaN   \n",
       "2300572     NaN    NaN        NaN    NaN   \n",
       "2300573     NaN    NaN        NaN    NaN   \n",
       "2300574     NaN    NaN        NaN    NaN   \n",
       "\n",
       "                                                     CLNVI CLNDISDBINCL  \\\n",
       "0                                                      NaN          NaN   \n",
       "1                                                      NaN          NaN   \n",
       "2                                                      NaN          NaN   \n",
       "3                                                      NaN          NaN   \n",
       "4                                                      NaN          NaN   \n",
       "...                                                    ...          ...   \n",
       "2300570                                   OMIM:110300.0001          NaN   \n",
       "2300571                  ClinGen:CA123258|OMIM:173445.0001          NaN   \n",
       "2300572  ClinGen:CA114051|OMIM:613236.0003|UniProtKB:B7...          NaN   \n",
       "2300573  ClinGen:CA114049|OMIM:613236.0002|UniProtKB:B7...          NaN   \n",
       "2300574  ClinGen:CA114053|OMIM:613236.0004|UniProtKB:B7...          NaN   \n",
       "\n",
       "        CLNDNINCL CLNSIGINCL DBVARID  \n",
       "0             NaN        NaN     NaN  \n",
       "1             NaN        NaN     NaN  \n",
       "2             NaN        NaN     NaN  \n",
       "3             NaN        NaN     NaN  \n",
       "4             NaN        NaN     NaN  \n",
       "...           ...        ...     ...  \n",
       "2300570       NaN        NaN     NaN  \n",
       "2300571       NaN        NaN     NaN  \n",
       "2300572       NaN        NaN     NaN  \n",
       "2300573       NaN        NaN     NaN  \n",
       "2300574       NaN        NaN     NaN  \n",
       "\n",
       "[2300575 rows x 28 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1ea2993d-6af3-4ec5-8518-e7c577e193b2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chrom 0\n",
      "Pos 0\n",
      "Id 0\n",
      "Ref 0\n",
      "Alt 0\n",
      "Qual 0\n",
      "Filter 0\n",
      "ALLELEID 0\n",
      "CLNDISDB 719\n",
      "CLNDN 719\n",
      "CLNHGVS 19\n",
      "CLNREVSTAT 0\n",
      "CLNSIG 719\n",
      "CLNVC 0\n",
      "CLNVCSO 0\n",
      "GENEINFO 453\n",
      "MC 14169\n",
      "ORIGIN 3775\n",
      "RS 1465846\n",
      "AF_EXAC 1993029\n",
      "AF_ESP 2131141\n",
      "CLNSIGCONF 2196618\n",
      "AF_TGP 2107477\n",
      "CLNVI 1904506\n",
      "CLNDISDBINCL 2298887\n",
      "CLNDNINCL 2298887\n",
      "CLNSIGINCL 2298887\n",
      "DBVARID 2300423\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Id</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Qual</th>\n",
       "      <th>Filter</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>...</th>\n",
       "      <th>RS</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>CLNSIGCONF</th>\n",
       "      <th>AF_TGP</th>\n",
       "      <th>CLNVI</th>\n",
       "      <th>CLNDISDBINCL</th>\n",
       "      <th>CLNDNINCL</th>\n",
       "      <th>CLNSIGINCL</th>\n",
       "      <th>DBVARID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4985</th>\n",
       "      <td>1</td>\n",
       "      <td>1455576</td>\n",
       "      <td>1343247</td>\n",
       "      <td>G</td>\n",
       "      <td>GGCCGAGGCCCGGGCGCGC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1334892</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0032931,MedGen:C5394137,OMIM:61881...</td>\n",
       "      <td>Pontocerebellar_hypoplasia,_hypotonia,_and_res...</td>\n",
       "      <td>1343267:Likely_pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13987</th>\n",
       "      <td>1</td>\n",
       "      <td>8045031</td>\n",
       "      <td>487089</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>22107</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>74315354</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OMIM:602533.0006|UniProtKB:Q99497#VAR_034801</td>\n",
       "      <td>MONDO:MONDO:0011658,MedGen:C1853445,OMIM:60632...</td>\n",
       "      <td>Autosomal_recessive_early-onset_Parkinson_dise...</td>\n",
       "      <td>446717:Pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15782</th>\n",
       "      <td>1</td>\n",
       "      <td>10035809</td>\n",
       "      <td>982549</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>970649</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1641792929</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0018998,MeSH:D057130,MedGen:C03395...</td>\n",
       "      <td>Leber_congenital_amaurosis</td>\n",
       "      <td>982550:Pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41011</th>\n",
       "      <td>1</td>\n",
       "      <td>25611101</td>\n",
       "      <td>635094</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>622837</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>199509194</td>\n",
       "      <td>0.00665</td>\n",
       "      <td>0.02480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.14217</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.|.</td>\n",
       "      <td>Partial_RhD|altered_RHD_phenotype</td>\n",
       "      <td>634995:not_provided|1185597:Affects</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41017</th>\n",
       "      <td>1</td>\n",
       "      <td>25617206</td>\n",
       "      <td>635095</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>622838</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>113982491</td>\n",
       "      <td>0.00665</td>\n",
       "      <td>0.02452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.|.</td>\n",
       "      <td>Partial_RhD|altered_RHD_phenotype</td>\n",
       "      <td>634995:not_provided|634996:not_provided|118559...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2295933</th>\n",
       "      <td>X</td>\n",
       "      <td>153762251</td>\n",
       "      <td>1723203</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1780260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...</td>\n",
       "      <td>Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...</td>\n",
       "      <td>1722757:Likely_pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2296005</th>\n",
       "      <td>X</td>\n",
       "      <td>153762653</td>\n",
       "      <td>242750</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>38393</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>267606836</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OMIM:305900.0032</td>\n",
       "      <td>MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...</td>\n",
       "      <td>Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...</td>\n",
       "      <td>10392:Likely_pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2296057</th>\n",
       "      <td>X</td>\n",
       "      <td>153763489</td>\n",
       "      <td>1723202</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1780230</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...</td>\n",
       "      <td>Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...</td>\n",
       "      <td>1722723:Likely_pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2296060</th>\n",
       "      <td>X</td>\n",
       "      <td>153763493</td>\n",
       "      <td>1723201</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1780229</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...</td>\n",
       "      <td>Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...</td>\n",
       "      <td>1722723:Likely_pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2296078</th>\n",
       "      <td>X</td>\n",
       "      <td>153763551</td>\n",
       "      <td>242797</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>25431</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>267606835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OMIM:305900.0032</td>\n",
       "      <td>MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...</td>\n",
       "      <td>Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...</td>\n",
       "      <td>10392:Likely_pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>719 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chrom        Pos       Id Ref                  Alt Qual Filter  \\\n",
       "4985        1    1455576  1343247   G  GGCCGAGGCCCGGGCGCGC    .      .   \n",
       "13987       1    8045031   487089   G                    A    .      .   \n",
       "15782       1   10035809   982549   G                    A    .      .   \n",
       "41011       1   25611101   635094   G                    T    .      .   \n",
       "41017       1   25617206   635095   C                    T    .      .   \n",
       "...       ...        ...      ...  ..                  ...  ...    ...   \n",
       "2295933     X  153762251  1723203   G                    T    .      .   \n",
       "2296005     X  153762653   242750   G                    A    .      .   \n",
       "2296057     X  153763489  1723202   C                    A    .      .   \n",
       "2296060     X  153763493  1723201   C                    A    .      .   \n",
       "2296078     X  153763551   242797   G                    C    .      .   \n",
       "\n",
       "        ALLELEID CLNDISDB CLNDN  ...          RS  AF_EXAC   AF_ESP CLNSIGCONF  \\\n",
       "4985     1334892      NaN   NaN  ...         NaN      NaN      NaN        NaN   \n",
       "13987      22107      NaN   NaN  ...    74315354  0.00002      NaN        NaN   \n",
       "15782     970649      NaN   NaN  ...  1641792929      NaN      NaN        NaN   \n",
       "41011     622837      NaN   NaN  ...   199509194  0.00665  0.02480        NaN   \n",
       "41017     622838      NaN   NaN  ...   113982491  0.00665  0.02452        NaN   \n",
       "...          ...      ...   ...  ...         ...      ...      ...        ...   \n",
       "2295933  1780260      NaN   NaN  ...         NaN      NaN      NaN        NaN   \n",
       "2296005    38393      NaN   NaN  ...   267606836  0.00001      NaN        NaN   \n",
       "2296057  1780230      NaN   NaN  ...         NaN      NaN      NaN        NaN   \n",
       "2296060  1780229      NaN   NaN  ...         NaN      NaN      NaN        NaN   \n",
       "2296078    25431      NaN   NaN  ...   267606835      NaN      NaN        NaN   \n",
       "\n",
       "          AF_TGP                                         CLNVI  \\\n",
       "4985         NaN                                           NaN   \n",
       "13987        NaN  OMIM:602533.0006|UniProtKB:Q99497#VAR_034801   \n",
       "15782        NaN                                           NaN   \n",
       "41011    0.14217                                           NaN   \n",
       "41017    0.02157                                           NaN   \n",
       "...          ...                                           ...   \n",
       "2295933      NaN                                           NaN   \n",
       "2296005      NaN                              OMIM:305900.0032   \n",
       "2296057      NaN                                           NaN   \n",
       "2296060      NaN                                           NaN   \n",
       "2296078      NaN                              OMIM:305900.0032   \n",
       "\n",
       "                                              CLNDISDBINCL  \\\n",
       "4985     MONDO:MONDO:0032931,MedGen:C5394137,OMIM:61881...   \n",
       "13987    MONDO:MONDO:0011658,MedGen:C1853445,OMIM:60632...   \n",
       "15782    MONDO:MONDO:0018998,MeSH:D057130,MedGen:C03395...   \n",
       "41011                                                  .|.   \n",
       "41017                                                  .|.   \n",
       "...                                                    ...   \n",
       "2295933  MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...   \n",
       "2296005  MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...   \n",
       "2296057  MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...   \n",
       "2296060  MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...   \n",
       "2296078  MONDO:MONDO:0010480,MedGen:C2720289,OMIM:30090...   \n",
       "\n",
       "                                                 CLNDNINCL  \\\n",
       "4985     Pontocerebellar_hypoplasia,_hypotonia,_and_res...   \n",
       "13987    Autosomal_recessive_early-onset_Parkinson_dise...   \n",
       "15782                           Leber_congenital_amaurosis   \n",
       "41011                    Partial_RhD|altered_RHD_phenotype   \n",
       "41017                    Partial_RhD|altered_RHD_phenotype   \n",
       "...                                                    ...   \n",
       "2295933  Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...   \n",
       "2296005  Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...   \n",
       "2296057  Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...   \n",
       "2296060  Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...   \n",
       "2296078  Anemia,_nonspherocytic_hemolytic,_due_to_G6PD_...   \n",
       "\n",
       "                                                CLNSIGINCL DBVARID  \n",
       "4985                             1343267:Likely_pathogenic     NaN  \n",
       "13987                                    446717:Pathogenic     NaN  \n",
       "15782                                    982550:Pathogenic     NaN  \n",
       "41011                  634995:not_provided|1185597:Affects     NaN  \n",
       "41017    634995:not_provided|634996:not_provided|118559...     NaN  \n",
       "...                                                    ...     ...  \n",
       "2295933                          1722757:Likely_pathogenic     NaN  \n",
       "2296005                            10392:Likely_pathogenic     NaN  \n",
       "2296057                          1722723:Likely_pathogenic     NaN  \n",
       "2296060                          1722723:Likely_pathogenic     NaN  \n",
       "2296078                            10392:Likely_pathogenic     NaN  \n",
       "\n",
       "[719 rows x 28 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "for column_name in df.columns:\n",
    "    sum_na = df[column_name].isna().sum()\n",
    "    print(column_name, sum_na)\n",
    "df[\"CLNREVSTAT\"].unique()\n",
    "rows_with_nan = df[df[\"CLNSIG\"].isna()]\n",
    "rows_with_nan\n",
    "\n",
    "# nan values because of https://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/:\n",
    "# This value may not be submitted. It is used in the file variant_summary.txt.gz in the path  https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/. This file reports  '-' in the ClinicalSignificance column for an allele that was submitted to ClinVar only in combination with another allele (e.g.a submission with an interpretation of a haplotype or a compound heterozygote) and was not interpreted explictly.  ClinVar thus has no interpretation specific to that allele.  To find the the interpretation that includes that allele, you can query ClinVar by the AlleleID,\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88ea9a44-fb71-4731-9ae2-17125bff4e1a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([2273382.,    4704.,    3461.,    3040.,    2859.,    2762.,\n",
       "           2666.,    2597.,    2564.,    2540.]),\n",
       " array([    0. ,  2519.7,  5039.4,  7559.1, 10078.8, 12598.5, 15118.2,\n",
       "        17637.9, 20157.6, 22677.3, 25197. ]),\n",
       " <BarContainer object of 10 artists>)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "plt.hist(df[\"Ref\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c20fa220-1b64-4d3a-a5df-91f081d44bde",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Id</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Qual</th>\n",
       "      <th>Filter</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>...</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>CLNSIGCONF</th>\n",
       "      <th>AF_TGP</th>\n",
       "      <th>CLNVI</th>\n",
       "      <th>CLNDISDBINCL</th>\n",
       "      <th>CLNDNINCL</th>\n",
       "      <th>CLNSIGINCL</th>\n",
       "      <th>DBVARID</th>\n",
       "      <th>CLNREVSTAT_stars</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Chrom, Pos, Id, Ref, Alt, Qual, Filter, ALLELEID, CLNDISDB, CLNDN, CLNHGVS, CLNREVSTAT, CLNSIG, CLNVC, CLNVCSO, GENEINFO, MC, ORIGIN, RS, AF_EXAC, AF_ESP, CLNSIGCONF, AF_TGP, CLNVI, CLNDISDBINCL, CLNDNINCL, CLNSIGINCL, DBVARID, CLNREVSTAT_stars]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 29 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows_with_nan = df[df[\"CLNSIG\"].isna()]\n",
    "rows_with_nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4e149792-0346-4459-babe-19124eda5f90",
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"CLNSIG\"].value_counts()\n",
    "\n",
    "# https://www.ncbi.nlm.nih.gov/clinvar/docs/review_status/\n",
    "\n",
    "criteria_review = {\n",
    "    \"practice_guideline\" : 4,\n",
    "    \"reviewed_by_expert_panel\" : 3,\n",
    "    \"criteria_provided,_multiple_submitters,_no_conflicts\" : 2,\n",
    "    \"criteria_provided,_conflicting_interpretations\" : 1,\n",
    "    \"criteria_provided,_single_submitter\": 1,\n",
    "    \"no_assertion_criteria_provided\": 0,\n",
    "    \"no_interpretation_for_the_single_variant\": 0,\n",
    "    \"no_assertion_provided\": 0\n",
    "}\n",
    "    \n",
    "df[\"CLNREVSTAT_stars\"] = df[\"CLNREVSTAT\"].replace(criteria_review)\n",
    "\n",
    "# Filter just the ones that has 4, 3 or 2 stars\n",
    "df = df[df[\"CLNREVSTAT_stars\"].isin([4,3,2])]\n",
    "df[\"CLNSIG\"].value_counts()\n",
    "\n",
    "def df_to_vcf(df: pd.DataFrame, header: str) -> str:\n",
    "    \"\"\"\n",
    "    From a dataframe, it extracts the fields Chrom, Pos, Id, Ref, Alt, CLNSIG ref_num and alt_num\n",
    "    to create a VCF file\n",
    "    args:\n",
    "        df(pd.DataFrame): pandas dataframe containing previously defined columns\n",
    "        header(str): header of the vcf file that the output will contain\n",
    "    Return:\n",
    "        output_filename(str): vcf file path\n",
    "    \"\"\"\n",
    "    output_filename = \"/home/ocanal/Desktop/clinvar/hg19/filtered_vcf_file.vcf\"\n",
    "    \n",
    "    with open(output_filename, \"w\") as output_vcf:\n",
    "        output_vcf.write(header)\n",
    "        # defining constant values in vcf\n",
    "        qual = \".\"\n",
    "        filter = \".\"\n",
    "        for index, row in df.iterrows():\n",
    "            chrom = str(row[\"Chrom\"])\n",
    "            pos = str(row[\"Pos\"])\n",
    "            rs_id = str(row[\"Id\"])\n",
    "            ref = str(row[\"Ref\"])\n",
    "            alt = str(row[\"Alt\"])\n",
    "            clnsig = str(row[\"CLNSIG\"])\n",
    "            ref_num = str(row[\"ref_num\"])\n",
    "            alt_num = str(row[\"alt_num\"])\n",
    "            \n",
    "    \n",
    "            inf_field = f\"CLNSIG={clnsig};ref_num={ref_num};alt_num={alt_num}\"\n",
    "            \n",
    "            vcf_line = [chrom, pos, rs_id, ref, alt, qual, filter, inf_field]\n",
    "    \n",
    "            vcf_line = \"\\t\".join(vcf_line)\n",
    "            output_vcf.write(vcf_line + \"\\n\")\n",
    "        \n",
    "    return(output_filename)\n",
    "            \n",
    "        \n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d9cb3b73-2743-435a-b12a-d6207837b9e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4284"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"CLNSIG\"].unique()\n",
    "count = df['CLNSIG'].value_counts().get('Likely_pathogenic', 0)\n",
    "count\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2c691558-ab97-418d-b8af-40afec6a82f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_bening = df.loc[df[\"CLNSIG\"] == \"Benign\"]\n",
    "\n",
    "df_patho = df.loc[df[\"CLNSIG\"] == \"Pathogenic\"]\n",
    "df_uncertain = df.loc[df[\"CLNSIG\"] == \"Uncertain_significance\"]\n",
    "all_df = pd.concat([df_bening, df_patho, df_uncertain])\n",
    "# all_df.to_csv(\"clinvar.csv\", encoding=\"utf-8\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "222e507e-29a0-431c-b933-4be5cbab1dde",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CLNSIG\n",
       "Uncertain_significance    154309\n",
       "Benign                     35176\n",
       "Pathogenic                 28418\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_df\n",
    "all_df[\"CLNSIG\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "420b0bfc-3ea7-4e48-a4d5-d463adb92f93",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_92489/1417531837.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  known_df[\"Chrom\"] = known_df[\"Chrom\"].replace(replace_dict)\n"
     ]
    }
   ],
   "source": [
    "# data = pd.read_csv(\"/home/ocanal/Desktop/clinvar_parsed.vcf\", sep=\"\\t\")\n",
    "columns_to_keep = [\"Chrom\",\"Pos\", \"Id\", \"Ref\", \"RS\", \"Alt\", \"CLNSIG\"]\n",
    "known_df = all_df[columns_to_keep]\n",
    "\n",
    "known_df[\"Chrom\"].unique()\n",
    "replace_dict = {\n",
    "    \"X\": 23, \n",
    "    \"Y\": 24,\n",
    "    \"MT\": 25,\n",
    "    \"NW_003315925.1\": 26,\n",
    "    \"NW_003315947.1\": 27,\n",
    "    \"NT_113889.1\" : 28,\n",
    "    \"NT_167222.1\": 29\n",
    "}\n",
    "\n",
    "known_df[\"Chrom\"] = known_df[\"Chrom\"].replace(replace_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "39e6a481-b20c-4d53-b38f-6347ed2c3ce0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_92489/2816101851.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  known_df[\"ref_num\"] = known_df[\"Ref\"].replace(ref_alt_dict)\n",
      "/tmp/ipykernel_92489/2816101851.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  known_df[\"alt_num\"] = known_df[\"Alt\"].replace(ref_alt_dict)\n",
      "/tmp/ipykernel_92489/2816101851.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  known_df[\"cln_sig_num\"] = known_df[\"CLNSIG\"].replace(clnsig_dict)\n",
      "/tmp/ipykernel_92489/2816101851.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  known_df[\"ref_num\"] = known_df[\"ref_num\"].apply(lambda x: x if x in valid_values else -1)\n",
      "/tmp/ipykernel_92489/2816101851.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  known_df[\"alt_num\"] = known_df[\"alt_num\"].apply(lambda x: x if x in valid_values else -1)\n"
     ]
    }
   ],
   "source": [
    "ref_alt_dict = {\n",
    "    \"G\" : 1,\n",
    "    \"A\": 2,\n",
    "    \"C\": 3,\n",
    "    \"T\": 4\n",
    "}\n",
    "\n",
    "clnsig_dict = {\n",
    "    \"Pathogenic\": 0,\n",
    "    \"Uncertain_significance\": 1,\n",
    "    \"Benign\": 2\n",
    "}\n",
    "known_df[\"ref_num\"] = known_df[\"Ref\"].replace(ref_alt_dict)\n",
    "known_df[\"alt_num\"] = known_df[\"Alt\"].replace(ref_alt_dict)\n",
    "\n",
    "known_df[\"cln_sig_num\"] = known_df[\"CLNSIG\"].replace(clnsig_dict)\n",
    "valid_values = {1,2,3,4}\n",
    "known_df[\"ref_num\"] = known_df[\"ref_num\"].apply(lambda x: x if x in valid_values else -1)\n",
    "known_df[\"alt_num\"] = known_df[\"alt_num\"].apply(lambda x: x if x in valid_values else -1)\n",
    "# known_df.to_csv(\"/home/ocanal/Desktop/clinvar/hg19/clinvar_hg19_csv_last.csv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "cbb11463-39cd-46ee-aae0-c648ce6f9e95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Id</th>\n",
       "      <th>Ref</th>\n",
       "      <th>RS</th>\n",
       "      <th>Alt</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>ref_num</th>\n",
       "      <th>alt_num</th>\n",
       "      <th>cln_sig_num</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>1</td>\n",
       "      <td>898613</td>\n",
       "      <td>775485</td>\n",
       "      <td>C</td>\n",
       "      <td>61746776</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1144</th>\n",
       "      <td>1</td>\n",
       "      <td>949608</td>\n",
       "      <td>402986</td>\n",
       "      <td>G</td>\n",
       "      <td>1921</td>\n",
       "      <td>A</td>\n",
       "      <td>Benign</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1153</th>\n",
       "      <td>1</td>\n",
       "      <td>949654</td>\n",
       "      <td>1166514</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>Benign</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1216</th>\n",
       "      <td>1</td>\n",
       "      <td>955563</td>\n",
       "      <td>387476</td>\n",
       "      <td>G</td>\n",
       "      <td>539283387</td>\n",
       "      <td>C</td>\n",
       "      <td>Benign</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1233</th>\n",
       "      <td>1</td>\n",
       "      <td>955597</td>\n",
       "      <td>128310</td>\n",
       "      <td>G</td>\n",
       "      <td>115173026</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300272</th>\n",
       "      <td>25</td>\n",
       "      <td>15117</td>\n",
       "      <td>693828</td>\n",
       "      <td>T</td>\n",
       "      <td>1603225092</td>\n",
       "      <td>C</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300358</th>\n",
       "      <td>25</td>\n",
       "      <td>15437</td>\n",
       "      <td>235525</td>\n",
       "      <td>G</td>\n",
       "      <td>878853058</td>\n",
       "      <td>A</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300369</th>\n",
       "      <td>25</td>\n",
       "      <td>15467</td>\n",
       "      <td>618216</td>\n",
       "      <td>A</td>\n",
       "      <td>1569484723</td>\n",
       "      <td>G</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300400</th>\n",
       "      <td>25</td>\n",
       "      <td>15579</td>\n",
       "      <td>9683</td>\n",
       "      <td>A</td>\n",
       "      <td>207460002</td>\n",
       "      <td>G</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2300471</th>\n",
       "      <td>25</td>\n",
       "      <td>15848</td>\n",
       "      <td>377369</td>\n",
       "      <td>A</td>\n",
       "      <td>1057520206</td>\n",
       "      <td>G</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>217903 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chrom     Pos       Id Ref          RS Alt                  CLNSIG  \\\n",
       "976         1  898613   775485   C    61746776   T                  Benign   \n",
       "1144        1  949608   402986   G        1921   A                  Benign   \n",
       "1153        1  949654  1166514   A         NaN   G                  Benign   \n",
       "1216        1  955563   387476   G   539283387   C                  Benign   \n",
       "1233        1  955597   128310   G   115173026   T                  Benign   \n",
       "...       ...     ...      ...  ..         ...  ..                     ...   \n",
       "2300272    25   15117   693828   T  1603225092   C  Uncertain_significance   \n",
       "2300358    25   15437   235525   G   878853058   A  Uncertain_significance   \n",
       "2300369    25   15467   618216   A  1569484723   G  Uncertain_significance   \n",
       "2300400    25   15579     9683   A   207460002   G  Uncertain_significance   \n",
       "2300471    25   15848   377369   A  1057520206   G  Uncertain_significance   \n",
       "\n",
       "         ref_num  alt_num  cln_sig_num  \n",
       "976            3        4            2  \n",
       "1144           1        2            2  \n",
       "1153           2        1            2  \n",
       "1216           1        3            2  \n",
       "1233           1        4            2  \n",
       "...          ...      ...          ...  \n",
       "2300272        4        3            1  \n",
       "2300358        1        2            1  \n",
       "2300369        2        1            1  \n",
       "2300400        2        1            1  \n",
       "2300471        2        1            1  \n",
       "\n",
       "[217903 rows x 10 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "known_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "dd2c7160-2fca-4fd2-8581-e8aba8780755",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/ocanal/Desktop/clinvar/hg19/filtered_vcf_file.vcf'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_df\n",
    "df_to_vcf(known_df, \"#CHROM\\tPOS\\tID\\tREF\\tALT\\tQUAL\\tFILTER\\tINFO\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f36cef17-ddcb-4a7e-b2c3-3e6a03ffb159",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c65f4007-0470-4654-95aa-e04cee59bb07",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_70875/2731417348.py:12: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  known_df[\"Chrom\"] = known_df[\"Chrom\"].replace(replace_dict)\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "bd515acc-1d43-4b2d-95cc-55bf60ced9c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_70875/2422956088.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  known_df[\"ref_num\"] = known_df[\"Ref\"].replace(ref_alt_dict)\n",
      "/tmp/ipykernel_70875/2422956088.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  known_df[\"alt_num\"] = known_df[\"Alt\"].replace(ref_alt_dict)\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'ref_num'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/pandas/core/indexes/base.py:3790\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3789\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 3790\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3791\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32mindex.pyx:152\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mindex.pyx:181\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:7080\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:7088\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'ref_num'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[14], line 11\u001b[0m\n\u001b[1;32m      8\u001b[0m known_df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124malt_num\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m known_df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mAlt\u001b[39m\u001b[38;5;124m\"\u001b[39m]\u001b[38;5;241m.\u001b[39mreplace(ref_alt_dict)\n\u001b[1;32m     10\u001b[0m valid_values \u001b[38;5;241m=\u001b[39m {\u001b[38;5;241m1\u001b[39m,\u001b[38;5;241m2\u001b[39m,\u001b[38;5;241m3\u001b[39m,\u001b[38;5;241m4\u001b[39m}\n\u001b[0;32m---> 11\u001b[0m known_df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mref_num\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[43mdf\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mref_num\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241m.\u001b[39mapply(\u001b[38;5;28;01mlambda\u001b[39;00m x: x \u001b[38;5;28;01mif\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m valid_values \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m)\n\u001b[1;32m     12\u001b[0m known_df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124malt_num\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m df[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124malt_num\u001b[39m\u001b[38;5;124m\"\u001b[39m]\u001b[38;5;241m.\u001b[39mapply(\u001b[38;5;28;01mlambda\u001b[39;00m x: x \u001b[38;5;28;01mif\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m valid_values \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m)\n\u001b[1;32m     13\u001b[0m known_df\u001b[38;5;241m.\u001b[39mto_csv(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/home/ocanal/Desktop/clinvar/hg19/clinvar_hg19_csv_last.csv\u001b[39m\u001b[38;5;124m\"\u001b[39m, sep\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/pandas/core/frame.py:3896\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3894\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mnlevels \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m   3895\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_getitem_multilevel(key)\n\u001b[0;32m-> 3896\u001b[0m indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3897\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n\u001b[1;32m   3898\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m [indexer]\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/pandas/core/indexes/base.py:3797\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3792\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(casted_key, \u001b[38;5;28mslice\u001b[39m) \u001b[38;5;129;01mor\u001b[39;00m (\n\u001b[1;32m   3793\u001b[0m         \u001b[38;5;28misinstance\u001b[39m(casted_key, abc\u001b[38;5;241m.\u001b[39mIterable)\n\u001b[1;32m   3794\u001b[0m         \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28many\u001b[39m(\u001b[38;5;28misinstance\u001b[39m(x, \u001b[38;5;28mslice\u001b[39m) \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m casted_key)\n\u001b[1;32m   3795\u001b[0m     ):\n\u001b[1;32m   3796\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m InvalidIndexError(key)\n\u001b[0;32m-> 3797\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[1;32m   3798\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m   3799\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[1;32m   3800\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[1;32m   3801\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[1;32m   3802\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[0;31mKeyError\u001b[0m: 'ref_num'"
     ]
    }
   ],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
